File: Visual Studio/C#/Eruru.MVVM/Controls/MVVMComboBox.cs
using System.Windows.Forms;

namespace Eruru.MVVM {

	public class MVVMComboBox : MVVMItemsControl {

		new readonly ComboBox Control;

		public MVVMComboBox (ComboBox control) : base (control) {
			Control = control;
		}
		public MVVMComboBox (MVVMView view, ComboBox control) : base (view, control) {
			Control = control;
		}

		protected override void Add (object value) {
			Control.Items.Add (value);
		}

		protected override void RemoveAt (int index) {
			Control.Items.RemoveAt (index);
		}

		protected override void Replace (int index, object value) {
			Control.Items[index] = value;
		}

		protected override void Move (int oldIndex, int newIndex) {
			Control.Items.Swap (oldIndex, newIndex);
		}

		protected override void Clear () {
			Control.Items.Clear ();
		}

	}

}

File: IntelliJ IDEA/Java/src/com/eruru/mvvm/swing/MVVMComboBox.java
package com.eruru.mvvm.swing;

import javax.swing.*;

public class MVVMComboBox extends MVVMItemsControl {

	private final JComboBox control;

	public MVVMComboBox (JComboBox control) {
		super (control);
		this.control = control;
	}

	public JComboBox getControl () {
		return control;
	}

	@Override
	protected Object convert (Object value) {
		return value;
	}

	@Override
	protected void add (Object value, int index) {
		control.insertItemAt (value, index);
	}

	@Override
	protected void add (Object value) {
		control.addItem (value);
	}

	@Override
	protected void removeAt (int index) {
		control.removeItemAt (index);
	}

	@Override
	protected void replace (int index, Object value) {
		control.removeItemAt (index);
		control.insertItemAt (value, index);
	}

	@Override
	public void move (int oldIndex, int newIndex) {
		Object value = control.getItemAt (oldIndex);
		control.removeItemAt (oldIndex);
		control.insertItemAt (value, newIndex);
	}

	@Override
	protected void reset () {
		control.removeAllItems ();
	}

}